src/lightning/data/streaming/dataset.py

```diff
@@ -68,12 +73,24 @@ def __init__(
 
         self.item_loader = item_loader
         self.shuffle: bool = shuffle
-        self.drop_last = drop_last
+        self.distributed_env = _DistributedEnv.detect()
+
+        if self.distributed_env.world_size > 1:
+            if drop_last is None or drop_last is True:
+                self.drop_last = True
+            else:
+                logger.warn(
+                    "You are running in a distributed environment and drop_last needs to be False. "
+                    "This can cause your training to hang if you are relying on distributed collectives."
+                )
+                self.drop_last = drop_last
+        else:
+            self.drop_last = drop_last or False
```

Comment:
```suggestion
        if self.distributed_env.world_size > 1:
            if drop_last is False:
                logger.warn(
                    "You are running in a distributed environment but `drop_last=False`. "
                    "This can cause your training to hang if you are relying on distributed collectives."
                )
            else:
                drop_last = True

        self.drop_last = drop_last or False
```

small change in warning message and can it be on rank zero?